server/core/src/irods_server_control_plane.cpp

```diff
@@ -1158,22 +999,22 @@ namespace irods {
         if ( !ret.ok() ) {
             irods::log( PASS( ret ) );
             return PASS( ret );
-
         }
 
         // add safeguards - if server is paused only allow a resume call
         server_state& s = server_state::instance();
         std::string the_server_state = s();
-        if ( server_state::PAUSED == the_server_state &&
-                SERVER_CONTROL_RESUME != cmd_name ) {
+        if ( server_state::PAUSED == the_server_state && SERVER_CONTROL_RESUME != cmd_name ) {
             _output = SERVER_PAUSED_ERROR;
             return SUCCESS();
         }
 
-        // the icat needs to be notified first in certain
+        resolve_hostnames_using_hosts_config(cmd_hosts);
```

Comment:
this line is the primary solution to #5700.

this line replaces hostnames in `cmd_hosts` with the name obtained from the hosts_config.json. also, `validate_host_list` likely does not need to call `resolve_hostname` due to this line.